src/compiler/transformers/classFields.ts

```diff
@@ -618,7 +618,11 @@ namespace ts {
 
         function transformConstructorBody(node: ClassDeclaration | ClassExpression, constructor: ConstructorDeclaration | undefined, isDerivedClass: boolean) {
             const useDefineForClassFields = context.getCompilerOptions().useDefineForClassFields;
-            const properties = getProperties(node, /*requireInitializer*/ !useDefineForClassFields, /*isStatic*/ false)
+            let properties = getProperties(node, /*requireInitializer*/ false, /*isStatic*/ false)
+            if (!useDefineForClassFields) {
+                properties = filter(properties, property => !!property.initializer || isPrivateIdentifier(property.name))

```

Comment:
nice catch here!

Reply:
thanks! Took me a bit to find where things went wrong.